src/core/src/package_managers/YumPackageManager.py

```diff
@@ -347,26 +386,249 @@ def get_package_size(self, output):
+        """ Initializes all generic auto OS update variables with the config values for yum cron service """
+        self.os_patch_configuration_settings_file_path = self.yum_cron_configuration_settings_file_path
+        self.download_updates_identifier_text = self.yum_cron_download_updates_identifier_text
+        self.apply_updates_identifier_text = self.yum_cron_apply_updates_identifier_text
+        self.auto_update_config_pattern_match_text = self.yum_cron_config_pattern_match_text
+        self.enable_on_reboot_check_cmd = self.yum_cron_enable_on_reboot_check_cmd
+
+    def __init_auto_update_for_dnf_automatic(self):
+        """ Initializes all generic auto OS update variables with the config values for dnf automatic service """
+        self.os_patch_configuration_settings_file_path = self.dnf_automatic_configuration_file_path
+        self.download_updates_identifier_text = self.dnf_automatic_download_updates_identifier_text
+        self.apply_updates_identifier_text = self.dnf_automatic_apply_updates_identifier_text
+        self.auto_update_config_pattern_match_text = self.dnf_automatic_config_pattern_match_text
+        self.enable_on_reboot_check_cmd = self.dnf_automatic_enable_on_reboot_check_cmd
 
     def disable_auto_os_update(self):
-        """ Disables auto OS updates on the machine only if they are enabled and logs the default settings the machine comes with """
-        # NOTE: Implementation pending
-        pass
+        """ Disables auto OS updates on the machine only if they are enable_on_reboot and logs the default settings the machine comes with """
+        try:
+            self.composite_logger.log_debug("Disabling auto OS updates in all identified services...")
+            for os_auto_update_service in self.auto_os_update_service_list:
+                # todo: Better way to implement this? Keeping functions separate since not all services have the same steps to disable and identifying the service specific commands is easier this way
+                if os_auto_update_service == self.yum_cron:
+                    self.disable_auto_os_update_for_yum_cron()
+                elif os_auto_update_service == self.dnf_automatic:
+                    self.disable_auto_os_update_for_dnf_automatic()
+                # elif os_auto_update_service == self.packagekit:
+                #     self.disable_auto_os_update_for_packagekit()
+                else:
+                    self.composite_logger.log_error("Disabling auto updates is not implemented for the service. Please check if this is required")
+            self.composite_logger.log_debug("Successfully disabled auto OS updates")
 
-    def is_image_default_patch_configuration_backup_valid(self, image_default_patch_configuration_backup):
-        return True
+        except Exception as error:
+            self.composite_logger.log_error("Could not disable auto OS updates. [Error={0}]".format(repr(error)))
+            raise
```

Comment:
Please make sure that a failure to configure does not torpedo the execution. This is covered in the configure patching class but sanity-check.